### PR TITLE
Fix modal dialogs

### PR DESCRIPTION
**Pull request by TheOneRing**
Eliminate Utility::setModal and rather call ->open() on all dialogs

    Utility::setModal was also setting Qt::ApplicationModal however
    it appears Qt never cared and always set Qt::WindowModal.
    https://github.com/qt/qtbase/blame/4ee4fc18b4067b90efa46ca9baba74f53b54d9ec/src/widgets/dialogs/qdialog.cpp#L543

    Setting Qt::Sheet is also handled in open();

    Fixes: #10621

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-ocis</strong> https://drone.owncloud.com/owncloud/client/14803/7/1

<details>


</details>

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14803/6/1

<details>


</details>

File: changelog/unreleased/10621 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Bugfix: Ui freeze when multiple modal dialogs displayed on MacOS
```
Comment (fmoc) [inline changelog/unreleased/10621:1]:
```suggestion
Bugfix: UI freeze when multiple modal dialogs displayed on macOS
```

File: changelog/unreleased/10621 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+Due to incorrect handling of the dialogs the ui could freeze.
```
Comment (fmoc) [inline changelog/unreleased/10621:3]:
```suggestion
Due to incorrect handling of the dialogs, the UI could freeze.
```

File: src/gui/loginrequireddialog/basicloginwidget.cpp [hunk 3]
```diff
@@ -48,11 +48,6 @@ BasicLoginWidget::BasicLoginWidget(QWidget *parent)
         _ui->usernameLineEdit->setPlaceholderText(Theme::instance()->userIDHint());
     }
 
-    qDebug() << _ui->usernameLabel->isEnabled();
```
Comment (fmoc) [inline src/gui/loginrequireddialog/basicloginwidget.cpp:51]: Cleanups like these should go into their own commit.

File: src/gui/tlserrordialog.cpp [hunk 4]
```diff
@@ -41,12 +41,7 @@ TlsErrorDialog::TlsErrorDialog(const QList<QSslError> &sslErrors, const QString
     connect(_ui->buttonBox, &QDialogButtonBox::accepted, this, [this]() {
         accept();
     });
-    connect(_ui->buttonBox, &QDialogButtonBox::rejected, this, [this]() {
-        reject();
-    });
-
-    // of course, we require an answer from the user, they may not proceed with anything else
-    setModal(true);
+    connect(_ui->buttonBox, &QDialogButtonBox::rejected, this, [this]() { reject(); });
```
Comment (fmoc) [inline src/gui/tlserrordialog.cpp:44]: Can't you connect to `QDialog::reject`?

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/tlserrordialog.cpp (hunk 4 above)
Comment (TheOneRing) [reply src/gui/tlserrordialog.cpp]: Yes, btw thats your code
